compiler-core/src/language_server/engine.rs

```diff
@@ -95,6 +97,7 @@ where
             progress_reporter,
             compiler,
             paths,
+            hex_deps: std::collections::HashSet::new(),
```

Comment:
The hex deps should be set immediately here as what they are is known even before compilation.